In [1]:
#It searches for the module initially in the local scope import. 
#The value returned by the function are then reflected in the output of the initial code.
import pandas as pd
import matplotlib.pyplot as plt

In [2]:
#pandas function to read csv files and do operations on it
df = pd.read_csv("./PCOS_raw_data.csv")

In [3]:
#returns the first 5 rows of the dataframe
df.head()

Sl No  Patient File No  PCOS (Y/N)  Age (yrs)  Weight (kg)  Height (cm)  \
0      1                1           0         28         44.6        152.0   
1      2                2           0         36         65.0        161.5   
2      3                3           1         33         68.8        165.0   
3      4                4           0         37         65.0        148.0   
4      5                5           0         25         52.0        161.0   

    BMI  Blood Group  Pulse rate (bpm)  RR (breaths/min)  ...  Pimples (Y/N)  \
0  19.3           15                78                22  ...              0   
1  24.9           15                74                20  ...              0   
2  25.3           11                72                18  ...              1   
3  29.7           13                72                20  ...              0   
4  20.1           11                72                18  ...              0   

   Fast Food (Y/N)  Reg.Exercise (Y/N)  BP_Systolic (mmHg)  \
0              1.0                   0                 110   
1              0.0                   0                 120   
2              1.0                   0                 120   
3              0.0                   0                 120   
4              0.0                   0                 120   

   BP_Diastolic (mmHg)  Follicle No (L)  Follicle No (R)  \
0                   80                3                3   
1                   70                3                5   
2                   80               13               15   
3                   70                2                2   
4                   80                3                4   

   Avg. F size (L) (mm)  Avg. F size (R) (mm)  Endometrium (mm)  
0                  18.0                  18.0               8.5  
1                  15.0                  14.0               3.7  
2                  18.0                  20.0              10.0  
3                  15.0                  14.0               7.5  
4                  16.0                  14.0               7.0  

[5 rows x 44 columns]

FINDINGS: Our dataset consists of 44 attributes.

In [4]:
# Quick idea of how big our data is
df.shape

(541, 44)

FINDINGS: Our dataset consists of 541 observations and 44 attributes.

In [5]:
#used to view some basic statistical details like percentile, mean, std etc.
df.describe()

Sl No  Patient File No  PCOS (Y/N)   Age (yrs)  Weight (kg)  \
count  541.000000       541.000000  541.000000  541.000000   541.000000   
mean   271.000000       271.000000    0.327172   31.430684    59.637153   
std    156.317519       156.317519    0.469615    5.411006    11.028287   
min      1.000000         1.000000    0.000000   20.000000    31.000000   
25%    136.000000       136.000000    0.000000   28.000000    52.000000   
50%    271.000000       271.000000    0.000000   31.000000    59.000000   
75%    406.000000       406.000000    1.000000   35.000000    65.000000   
max    541.000000       541.000000    1.000000   48.000000   108.000000   

       Height (cm)         BMI  Blood Group  Pulse rate (bpm)  \
count   541.000000  541.000000   541.000000        541.000000   
mean    156.484835   24.313124    13.802218         73.247689   
std       6.033545    4.051122     1.840812          4.430285   
min     137.000000   12.400000    11.000000         13.000000   
25%     152.000000   21.700000    13.000000         72.000000   
50%     156.000000   24.200000    14.000000         72.000000   
75%     160.000000   26.600000    15.000000         74.000000   
max     180.000000   38.900000    18.000000         82.000000   

       RR (breaths/min)  ...  Pimples (Y/N)  Fast Food (Y/N)  \
count        541.000000  ...     541.000000       540.000000   
mean          19.243993  ...       0.489834         0.514815   
std            1.688629  ...       0.500359         0.500244   
min           16.000000  ...       0.000000         0.000000   
25%           18.000000  ...       0.000000         0.000000   
50%           18.000000  ...       0.000000         1.000000   
75%           20.000000  ...       1.000000         1.000000   
max           28.000000  ...       1.000000         1.000000   

       Reg.Exercise (Y/N)  BP_Systolic (mmHg)  BP_Diastolic (mmHg)  \
count          541.000000          541.000000           541.000000   
mean             0.247689          114.661738            76.927911   
std              0.432070            7.384556             5.574112   
min              0.000000           12.000000             8.000000   
25%              0.000000          110.000000            70.000000   
50%              0.000000          110.000000            80.000000   
75%              0.000000          120.000000            80.000000   
max              1.000000          140.000000           100.000000   

       Follicle No (L)  Follicle No (R)  Avg. F size (L) (mm)  \
count       541.000000       541.000000            541.000000   
mean          6.129390         6.641405             15.018115   
std           4.229294         4.436889              3.566839   
min           0.000000         0.000000              0.000000   
25%           3.000000         3.000000             13.000000   
50%           5.000000         6.000000             15.000000   
75%           9.000000        10.000000             18.000000   
max          22.000000        20.000000             24.000000   

       Avg. F size (R) (mm)  Endometrium (mm)  
count            541.000000        541.000000  
mean              15.451701          8.475915  
std                3.318848          2.165381  
min                0.000000          0.000000  
25%               13.000000          7.000000  
50%               16.000000          8.500000  
75%               18.000000          9.800000  
max               24.000000         18.000000  

[8 rows x 43 columns]

In [6]:
#used to print information about DataFrame including the index dtype and column dtypes, non-null values and memory usage
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 541 entries, 0 to 540
Data columns (total 44 columns):
 #   Column                 Non-Null Count  Dtype  
---  ------                 --------------  -----  
 0   Sl No                  541 non-null    int64  
 1   Patient File No        541 non-null    int64  
 2   PCOS (Y/N)             541 non-null    int64  
 3   Age (yrs)              541 non-null    int64  
 4   Weight (kg)            541 non-null    float64
 5   Height (cm)            541 non-null    float64
 6   BMI                    541 non-null    float64
 7   Blood Group            541 non-null    int64  
 8   Pulse rate (bpm)       541 non-null    int64  
 9   RR (breaths/min)       541 non-null    int64  
 10  Hb (g/dl)              541 non-null    float64
 11  Cycle (R/I)            541 non-null    int64  
 12  Cycle Length (days)    541 non-null    int64  
 13  Marriage Status (yrs)  540 non-null    float64
 14  Pregnant (Y/N)         541 non-null    int64  
 15  No. of

In [7]:
#helps in removing duplicates from the data frame
df = df.drop_duplicates()

In [8]:
#will give the column-wise sum of missing values
print(df.isnull().sum())

Sl No                    0
Patient File No          0
PCOS (Y/N)               0
Age (yrs)                0
Weight (kg)              0
Height (cm)              0
BMI                      0
Blood Group              0
Pulse rate (bpm)         0
RR (breaths/min)         0
Hb (g/dl)                0
Cycle (R/I)              0
Cycle Length (days)      0
Marriage Status (yrs)    1
Pregnant (Y/N)           0
No. of abortions         0
I  beta-HCG (mIU/mL)     0
II  beta-HCG (mIU/mL)    0
FSH (mIU/mL)             0
LH (mIU/mL)              0
FSH/LH                   0
Hip (inch)               0
Waist (inch)             0
Waist/Hip Ratio          0
TSH (mIU/L)              0
AMH (ng/mL)              0
PRL (ng/mL)              0
Vit D3 (ng/mL)           0
PRG (ng/mL)              0
RBS (mg/dl)              0
Weight Gain (Y/N)        0
Hair Growth(Y/N)         0
Skin Darkening (Y/N)     0
Hair Loss (Y/N)          0
Pimples (Y/N)            0
Fast Food (Y/N)          1
Reg.Exercise (Y/N)       0
B

FINDINGS: Marraige Status (yrs) and Fast Food (Y/N) show a null value

In [9]:
#return a list of the null values
df[df['Marriage Status (yrs)'].isnull()].index.tolist()

[458]

FINDINGS: Row 458 has a NULL value

In [10]:
#prints range of values
print(df.loc[455:460,:"Marriage Status (yrs)"])

     Sl No  Patient File No  PCOS (Y/N)  Age (yrs)  Weight (kg)  Height (cm)  \
455    456              456           1         32         52.0      150.000   
456    457              457           1         31         51.0      156.000   
457    458              458           1         31         38.0      158.496   
458    459              459           1         36         66.0      162.000   
459    460              460           0         36         55.0      154.000   
460    461              461           0         27         63.0      158.000   

      BMI  Blood Group  Pulse rate (bpm)  RR (breaths/min)  Hb (g/dl)  \
455  23.1           13                70                18        9.4   
456  21.0           17                78                20       11.1   
457  15.1           15                80                20       12.0   
458  25.1           15                72                20       11.0   
459  23.2           13                74                18       10.0   
4

FINDINGS: Row 458 has a NaN value

In [11]:
#remove/delete row 458 from dataset
df1 = df.drop(df.index[458])

#prints range of values
print(df1.loc[455:460,:])

     Sl No  Patient File No  PCOS (Y/N)  Age (yrs)  Weight (kg)  Height (cm)  \
455    456              456           1         32         52.0      150.000   
456    457              457           1         31         51.0      156.000   
457    458              458           1         31         38.0      158.496   
459    460              460           0         36         55.0      154.000   
460    461              461           0         27         63.0      158.000   

      BMI  Blood Group  Pulse rate (bpm)  RR (breaths/min)  ...  \
455  23.1           13                70                18  ...   
456  21.0           17                78                20  ...   
457  15.1           15                80                20  ...   
459  23.2           13                74                18  ...   
460  25.2           12                72                18  ...   

     Pimples (Y/N)  Fast Food (Y/N)  Reg.Exercise (Y/N)  BP_Systolic (mmHg)  \
455              1              0.0  

FINDINGS: Row 458 is no longer showing

In [12]:
#will give the column-wise sum of missing values
print(df1.isnull().sum())

Sl No                    0
Patient File No          0
PCOS (Y/N)               0
Age (yrs)                0
Weight (kg)              0
Height (cm)              0
BMI                      0
Blood Group              0
Pulse rate (bpm)         0
RR (breaths/min)         0
Hb (g/dl)                0
Cycle (R/I)              0
Cycle Length (days)      0
Marriage Status (yrs)    0
Pregnant (Y/N)           0
No. of abortions         0
I  beta-HCG (mIU/mL)     0
II  beta-HCG (mIU/mL)    0
FSH (mIU/mL)             0
LH (mIU/mL)              0
FSH/LH                   0
Hip (inch)               0
Waist (inch)             0
Waist/Hip Ratio          0
TSH (mIU/L)              0
AMH (ng/mL)              0
PRL (ng/mL)              0
Vit D3 (ng/mL)           0
PRG (ng/mL)              0
RBS (mg/dl)              0
Weight Gain (Y/N)        0
Hair Growth(Y/N)         0
Skin Darkening (Y/N)     0
Hair Loss (Y/N)          0
Pimples (Y/N)            0
Fast Food (Y/N)          1
Reg.Exercise (Y/N)       0
B

FINDINGS: Fast Food (Y/N) still shows a null value

In [13]:
#return a list of the null values
df1[df1['Fast Food (Y/N)'].isnull()].index.tolist()

[156]

FINDINGS: Row 156 has a NULL value

In [14]:
#prints range of values
print(df1.loc[155:160,:"Fast Food (Y/N)"])

     Sl No  Patient File No  PCOS (Y/N)  Age (yrs)  Weight (kg)  Height (cm)  \
155    156              156           0         26         49.3        158.0   
156    157              157           0         27         53.2        158.0   
157    158              158           0         34         64.0        158.0   
158    159              159           1         25         60.0        151.0   
159    160              160           0         32         64.0        158.0   
160    161              161           0         36         50.0        143.0   

      BMI  Blood Group  Pulse rate (bpm)  RR (breaths/min)  ...  PRL (ng/mL)  \
155  19.7           15                76                18  ...        44.27   
156  21.3           13                72                22  ...         7.16   
157  25.6           15                73                20  ...        24.89   
158  26.3           15                74                18  ...        25.04   
159  25.6           15                7

FINDINGS: Row 156 has a NaN value

In [15]:
#remove/delete row 156 from dataset
data = df1.drop(df1.index[156])

#prints range of values
print(data.loc[155:160,:])

     Sl No  Patient File No  PCOS (Y/N)  Age (yrs)  Weight (kg)  Height (cm)  \
155    156              156           0         26         49.3        158.0   
157    158              158           0         34         64.0        158.0   
158    159              159           1         25         60.0        151.0   
159    160              160           0         32         64.0        158.0   
160    161              161           0         36         50.0        143.0   

      BMI  Blood Group  Pulse rate (bpm)  RR (breaths/min)  ...  \
155  19.7           15                76                18  ...   
157  25.6           15                73                20  ...   
158  26.3           15                74                18  ...   
159  25.6           15                73                18  ...   
160  24.5           13                72                20  ...   

     Pimples (Y/N)  Fast Food (Y/N)  Reg.Exercise (Y/N)  BP_Systolic (mmHg)  \
155              0              0.0  

FINDINGS: Row 156 is no longer showing

In [16]:
#will give the column-wise sum of missing values
print(data.isnull().sum())

Sl No                    0
Patient File No          0
PCOS (Y/N)               0
Age (yrs)                0
Weight (kg)              0
Height (cm)              0
BMI                      0
Blood Group              0
Pulse rate (bpm)         0
RR (breaths/min)         0
Hb (g/dl)                0
Cycle (R/I)              0
Cycle Length (days)      0
Marriage Status (yrs)    0
Pregnant (Y/N)           0
No. of abortions         0
I  beta-HCG (mIU/mL)     0
II  beta-HCG (mIU/mL)    0
FSH (mIU/mL)             0
LH (mIU/mL)              0
FSH/LH                   0
Hip (inch)               0
Waist (inch)             0
Waist/Hip Ratio          0
TSH (mIU/L)              0
AMH (ng/mL)              0
PRL (ng/mL)              0
Vit D3 (ng/mL)           0
PRG (ng/mL)              0
RBS (mg/dl)              0
Weight Gain (Y/N)        0
Hair Growth(Y/N)         0
Skin Darkening (Y/N)     0
Hair Loss (Y/N)          0
Pimples (Y/N)            0
Fast Food (Y/N)          0
Reg.Exercise (Y/N)       0
B

FINDINGS: No more NULL values found

In [17]:
# Quick idea of how big our data is
data.shape

(539, 44)

FINDINGS: Our dataset now consists of 539 observations and 44 attributes.

In [18]:
#remove first two columns as they are not needed for our study
data.drop(columns=['Sl No','Patient File No'])

PCOS (Y/N)  Age (yrs)  Weight (kg)  Height (cm)   BMI  Blood Group  \
0             0         28         44.6      152.000  19.3           15   
1             0         36         65.0      161.500  24.9           15   
2             1         33         68.8      165.000  25.3           11   
3             0         37         65.0      148.000  29.7           13   
4             0         25         52.0      161.000  20.1           11   
..          ...        ...          ...          ...   ...          ...   
536           0         35         50.0      164.592  18.5           17   
537           0         30         63.2      158.000  25.3           15   
538           0         36         54.0      152.000  23.4           13   
539           0         27         50.0      150.000  22.2           15   
540           1         23         82.0      165.000  30.1           13   

     Pulse rate (bpm)  RR (breaths/min)  Hb (g/dl)  Cycle (R/I)  ...  \
0                  78                22      10.48            2  ...   
1                  74                20      11.70            2  ...   
2                  72                18      11.80            2  ...   
3                  72                20      12.00            2  ...   
4                  72                18      10.00            2  ...   
..                ...               ...        ...          ...  ...   
536                72                16      11.00            2  ...   
537                72                18      10.80            2  ...   
538                74                20      10.80            2  ...   
539                74                20      12.00            4  ...   
540                80                20      10.20            4  ...   

     Pimples (Y/N)  Fast Food (Y/N)  Reg.Exercise (Y/N)  BP_Systolic (mmHg)  \
0                0              1.0                   0                 110   
1                0              0.0                   0                 120   
2                1              1.0                   0                 120   
3                0              0.0                   0                 120   
4                0              0.0                   0                 120   
..             ...              ...                 ...                 ...   
536              0              0.0                   0                 110   
537              0              0.0                   0                 110   
538              0              0.0                   0                 110   
539              1              0.0                   0                 110   
540              1              1.0                   0                 120   

     BP_Diastolic (mmHg)  Follicle No (L)  Follicle No (R)  \
0                     80                3                3   
1                     70                3                5   
2                     80               13               15   
3                     70                2                2   
4                     80                3                4   
..                   ...              ...              ...   
536                   70                1                0   
537                   70                9                7   
538                   80                1                0   
539                   70                7                6   
540                   70                9               10   

     Avg. F size (L) (mm)  Avg. F size (R) (mm)  Endometrium (mm)  
0                    18.0                  18.0               8.5  
1                    15.0                  14.0               3.7  
2                    18.0                  20.0              10.0  
3                    15.0                  14.0               7.5  
4                    16.0                  14.0               7.0  
..                    ...                   ...               ...  
536                  17.5                  10.0               6.7  
5

FINDINGS: First to columns no longer display

In [19]:
#export your Pandas DataFrame to a CSV file
data.to_csv('PCOS_New.csv')